src/GithubFilesBundle/Service/GithubImport.php

```diff
@@ -0,0 +1,166 @@
+     *     "em" = @DI\Inject("doctrine.orm.entity_manager"),
+     * })
+     */
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @return EntityManager
+     */
+    protected function getEntityManager()
+    {
+        return $this->em;
+    }
+
+    /**
+     * Parse an Github document into a LsDoc/LsItem hierarchy
+     *
+     * @param array $lsdocKeys
+     * @param string $fileContent
+     */
+    public function parseGithubDocument($lsDocKeys, $lsItemKeys, $fileContent){
+        $csvContent = str_getcsv($fileContent, "\n");
+        $headers = array();
+        $content = array();
+        $tempContent = array();
+
+        foreach ($csvContent as $i => $row) {
+            $row = str_getcsv($row, ",");
+
+            if ($i == 0){
+                $headers = $row;
+                continue;
+            }
+
+            foreach($headers as $h => $col){
+                $tempContent[$col] = $row[$h];
+            }
+
+            array_push($content, $tempContent);
+            $tempContent = array();
+        }
+
+        $this->saveGithubDocument($lsDocKeys, $lsItemKeys, $content);
+    }
+
+    /**
+     * Save an Github document into a LsDoc/LsItem hierarchy
+     *
+     * @param array $lsdocKeys
+     * @param string $content
+     */
+    public function saveGithubDocument($lsDocKeys, $lsItemKeys, $content){
+        $em = $this->getEntityManager();
+        $lsDoc = new LsDoc();
+
+        foreach($content as $i => $row){
+            if($i < 1){
+                //create the new LsDoc
+                $lsDocIdentifier = Uuid::uuid5(Uuid::NAMESPACE_URL, $this->getValue($lsDocKeys['title'], $row))->toString();
+                $lsDoc->setIdentifier($lsDocIdentifier);
+                $lsDoc->setUri('local:'.$lsDocIdentifier);
+
+                if(empty($this->getValue($lsDocKeys['creator'], $row))){
+                    $lsDoc->setCreator('Imported from GitHub');
+                }else{
+                    $lsDoc->setCreator($this->getValue($lsDocKeys['creator'], $row));
+                }
+
+                $lsDoc->setTitle($this->getValue($lsDocKeys['title'], $row));
+
+                $lsDoc->setOfficialUri($this->getValue($lsDocKeys['officialSourceURL'], $row));
+                $lsDoc->setPublisher($this->getValue($lsDocKeys['publisher'], $row));
+                $lsDoc->setDescription($this->getValue($lsDocKeys['description'], $row));
+                $lsDoc->setVersion($this->getValue($lsDocKeys['version'], $row));
+                $lsDoc->setSubject($this->getValue($lsDocKeys['subject'], $row));
+                $lsDoc->setLanguage($this->getValue($lsDocKeys['language'], $row));
+                $lsDoc->setNote($this->getValue($lsDocKeys['notes'], $row));
+
+                $em->persist($lsDoc);
+                $em->flush();
+                continue;
+            }
+
+            $lsItem = $this->parseGithubStandard($lsDoc, $lsItemKeys, $row);
```

Comment:
`$lsItem` will always be null here as `parseGithubStandard` does not return a value.